deeplake/core/dataset/dataset.py

```diff
@@ -3202,6 +3209,32 @@ def _copy(
             else False,
         )
 
+        def _copy_tensor(sample_in, sample_out):
+            for tensor_name in dest_ds.tensors:
+                src = self[tensor_name]
+                if (
+                    unlink
+                    and src.is_link
+                    and (src.base_htype != "video" or deeplake.constants._UNLINK_VIDEOS)
+                ):
+                    if len(self.index) > 1:
+                        sample_out[tensor_name].extend(sample_in[tensor_name])
```

Comment:
would this work correctly? wouldn't sample_in[tensor_name] return sliced samples?

Reply:
it will call `.numpy` during serialization